JavaScript/JavaScriptSDK/appInsights.ts

```diff
@@ -301,7 +318,31 @@ module Microsoft.ApplicationInsights {
 
                 this.context.track(envelope);
             } catch (e) {
-                _InternalLogging.warnToConsole("trackTrace failed, trace will not be collected: " + Util.dump(e));
+                _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.WARNING, "trackTrace failed, trace will not be collected: " + Util.dump(e));
+            }
+        }
+
+        /**
+        * Log a page visit time
+        * @param    pageName    Name of page
+        * @param    pageVisitDuration Duration of visit to the page in milleseconds
+        */
+        public trackPageVisitTime(pageName: string, pageUrl: string, pageVisitTime: number) {
```

Comment:
At this time `trackPageVisitTime` appears to be a convenience shortcut for calling `trackMetric` with a predefined set of known and custom properties. Do we expect it to be used by _most_ AI customers?